### PR TITLE
Integration of Aqua Security

### PR DESCRIPTION
**Pull request by hrcornejo**
Based in PR: https://github.com/opendevstack/ods-jenkins-shared-library/pull/617

### CODE REVIEW

**Comment by michaelsauter** [comment]
I thought the scan was supposed to be included in the image build step?

**Comment by renedupont** [comment]
@michaelsauter I thought this happens here? https://github.com/opendevstack/ods-jenkins-shared-library/pull/661/files#diff-d8f79fbab3b26b1bf0c74c6c3009c1634bae992d20209ebd20fda6d53455513dR27

**Comment by michaelsauter** [comment]
@renedupont Ah, right, I looked in the wrong place (I expected a call to the service in the build stage). All good then ...

File: src/org/ods/component/ScanWithAquaStage.groovy [hunk 1]
```diff
@@ -10,7 +10,9 @@ import org.ods.util.ILogger
 @TypeChecked
 class ScanWithAquaStage extends Stage {
 
-    public final String STAGE_NAME = 'Aqua Security Scan'
+    static final String STAGE_NAME = 'Aqua Security Scan'
+    static final String AQUA_CONFIG_MAP_NAME = "aqua"
+    static final String AQUA_GENERAL_CONFIG_MAP_PROJECT = "ods"
```
[the remark this answers is not in the document]
Comment (hrcornejo) [reply src/org/ods/component/ScanWithAquaStage.groovy:15]: In that case, @renedupont  how I can get the name of namespace, to use it?

File: src/org/ods/component/ScanWithAquaStage.groovy [hunk 2]
```diff
@@ -135,4 +196,18 @@ class ScanWithAquaStage extends Stage {
...
 
+    private void notifyAquaProblem(String recipients = '', String message = '') {
+        String subject = "Build $context.componentId on project $context.projectId had some problems with Aqua!"
+        String body = "<p>$subject</p> " +
+            "<p>URL : <a href=\"$context.buildUrl\">$context.buildUrl</a></p> <ul>$message</ul>"
+
+        if (message) {
+            steps.emailext(
+                body: body, mimeType: 'text/html',
+                replyTo: '$script.DEFAULT_REPLYTO', subject: subject,
```
[the remark this answers is not in the document]
Comment (hrcornejo) [reply src/org/ods/component/ScanWithAquaStage.groovy:207]: With that value, uses the default conf in Jenkins:
image
So I hope that will be ok :)

File: src/org/ods/component/ScanWithAquaStage.groovy (hunk 1 above)
[the remark this answers is not in the document]
Comment (renedupont) [reply src/org/ods/component/ScanWithAquaStage.groovy:15]: Not sure right now tbh, maybe @michaelsauter has an idea?

[the remark this answers is not in the document]
Comment (metmajer) [reply src/org/ods/component/ScanWithAquaStage.groovy:15]: @michaelsauter ping ;-) and idea?

**Comment by metmajer** [comment]
@hrcornejo are you depending on a review or shall we turn this into a Draft until you finished the pending items on our Jira board?

File: src/org/ods/component/ScanWithAquaStage.groovy (hunk 1 above)
[the remark this answers is not in the document]
Comment (hrcornejo) [reply src/org/ods/component/ScanWithAquaStage.groovy]: Modified to use JENKINS_MASTER_OPENSHIFT_BUILD_NAMESPACE from config parameters of steps.

**Comment by hrcornejo** [comment]
> @hrcornejo are you depending on a review or shall we turn this into a Draft until you finished the pending items on our Jira board?

**Comment by hrcornejo** [comment]
@metmajer It can be changed to a Draft, I'll request a new review after finalize the rest

File: src/org/ods/services/BitbucketService.groovy [hunk 3]
```diff
@@ -309,6 +309,50 @@ class BitbucketService {
...
+                    label: 'Create Bitbucket Code Insight report via API',
+                    script: """curl \\
+                        --fail \\
+                        -sS \\
+                        --request PUT \\
+                        --header \"Authorization: Bearer ${token}\" \\
+                        --header \"Content-Type: application/json\" \\
+                        --data '${payload}' \\
+                        ${bitbucketUrl}/rest/insights/1.0/projects/${project}/\
+repos/${repo}/commits/${gitCommit}/reports/${gitCommit}"""
```
Comment (michaelsauter) [inline src/org/ods/services/BitbucketService.groovy:347]: I think the report key should be more specific, Atlassian says on this:
```
The report key should be a unique string chosen by the reporter and should be unique enough not to potentially clash with report keys from other reporters. We recommend using reverse DNS namespacing or a similar standard to ensure that collision is avoided.
```

See https://docs.atlassian.com/bitbucket-server/rest/7.13.0/bitbucket-code-insights-rest.html#idp9

Comment (hrcornejo) [reply src/org/ods/services/BitbucketService.groovy:347]: Well, now the key is the commit-SHA. Perhaps adding something like "aqua-report"? Eg: "1212121212-aqua-report"

Comment (michaelsauter) [reply src/org/ods/services/BitbucketService.groovy:347]: I think the key is meant to be something "stable" like `org.opendevstack.aquasec`. It's already connected to the commit anyway ...

Comment (hrcornejo) [reply src/org/ods/services/BitbucketService.groovy]: Modified to `org.opendevstack.aquasec`

**Review by jorge-romero — APPROVED** [review]

**Review by metmajer — APPROVED** [review]

**Pull request merged**